packages/edit-post/src/components/keyboard-shortcuts/index.js

```diff
@@ -200,6 +205,35 @@ function KeyboardShortcuts() {
 		setIsListViewOpened( ! isListViewOpened() )
 	);
 
+	useShortcut(
+		'core/block-editor/transform-paragraph-to-heading-1',
+		( event ) => handleTextLevelShortcut( event, 1 )
+	);
+	useShortcut(
+		'core/block-editor/transform-paragraph-to-heading-2',
+		( event ) => handleTextLevelShortcut( event, 2 )
+	);
+	useShortcut(
+		'core/block-editor/transform-paragraph-to-heading-3',
+		( event ) => handleTextLevelShortcut( event, 3 )
+	);
+	useShortcut(
+		'core/block-editor/transform-paragraph-to-heading-4',
+		( event ) => handleTextLevelShortcut( event, 4 )
+	);
+	useShortcut(
+		'core/block-editor/transform-paragraph-to-heading-5',
+		( event ) => handleTextLevelShortcut( event, 5 )
+	);
+	useShortcut(
+		'core/block-editor/transform-paragraph-to-heading-6',
+		( event ) => handleTextLevelShortcut( event, 6 )
+	);
```

Comment:
I think this could technically put this into a loop as well without breaking the rule of hooks (because the loop is based on a constant, it will always execute the same way every time).

You could then avoid having to export `handleTextLevelShortcut` from a hook, and have it work the same way as the other `useShortcut` calls where all the logic is inline.

Reply:
I felt I needed permissions to break the holy react rules